### PR TITLE
Improving data-sharing across Containers

### PR DESCRIPTION
**Pull request by Borales**
### Please check if the PR fulfills these requirements ###

- [x] The commit message follows our guidelines
- [x] Changelog have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**
Improvement

**What is the current behavior? (You can also link to an open issue here)**
When implementing endpoints for cache invalidation - there was no `cache` component exposed in the `EndpointManager` instance.

**Does this PR introduce a breaking change?**
No

### CODE REVIEW

File: packages/falcon-server/src/containers/EndpointContainer.ts [hunk 1]
```diff
@@ -21,6 +26,7 @@ export class EndpointContainer extends BaseContainer {
           return;
         }
         const endpointManager = new EndpointManagerClass({
+          cache: this.cache,
```
Comment (nimobeeren) [inline packages/falcon-server/src/containers/EndpointContainer.ts]: Perhaps I'm missing something but I don't think `this.cache` is defined here. The `cache` property was added to the `EndpointManager` class but this class does not inherit from it, nor do we set its value in the constructor.

Comment (Borales) [reply packages/falcon-server/src/containers/EndpointContainer.ts]: @nimobeeren `cache` is being passed on line 10 of this file. `EndpointManager` constructor argument has been updated as well:

Screenshot 2019-11-14 at 09 44 50

Comment (nimobeeren) [reply packages/falcon-server/src/containers/EndpointContainer.ts]: @Borales the constructor of `EndpointContainer` does take a `cache` parameter, but I don't see that we use it anywhere. Yet we try to access `this.cache` without ever setting it.

Comment (Borales) [reply packages/falcon-server/src/containers/EndpointContainer.ts]: @nimobeeren I guess you're asking about the constructor assignment, aren't you? 

```typescript
constructor(eventEmitter: EventEmitter2, protected cache: Cache) {

}
```

Having `protected` keyword in the constructor argument automatically does the assignment for us.

Comment (Borales) [reply packages/falcon-server/src/containers/EndpointContainer.ts]: Here's how it looks after being compiled - [Playground Link](http://www.typescriptlang.org/play/index.html?target=1#code/MYGwhgzhAEBiD29oG8CwAoaXrHgOwgBcAnAV2EPmIAoAHY+QgUwqYBNo8wBbJgLmhFiASzwBzAJQoM2WdBnYAvhkVA)

Comment (nimobeeren) [reply packages/falcon-server/src/containers/EndpointContainer.ts]: @Borales ah cool! I was looking for something like that but couldn't find it in the TS docs. That clears it up!

**Pull request merged**